test/framework/integration_framework/fake_peer/proposal_storage.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/integration_framework/fake_peer/proposal_storage.hpp"
+
+namespace integration_framework {
+  namespace fake_peer {
+
+    ProposalStorage::ProposalStorage()
+        : default_provider_([](auto &) { return boost::none; }) {}
+
+    OrderingProposalRequestResult ProposalStorage::getProposal(
+        const Round &round) const {
+      auto it = proposals_map_.find(round);
+      if (it != proposals_map_.end()) {
+        if (it->second) {
```

Comment:
might be a bit clearer:
`return it->second ? OrderingProposalRequestResult{*it->second}: boost::none;`

Reply:
the ternary operator wants both alternatives of  the same type, and accomplishing this makes the code way uglier than the current version, which is at least easily readable.